contracts/erc4626/AutoCompounder.sol

```diff
@@ -0,0 +1,219 @@
+        address _uniswapV2Router,
+        address _vault,
+        address _usdc,
+        string memory _name,
+        string memory _symbol
+    ) payable ERC20(_name, _symbol, ERC20(IERC4626(_vault).asset()).decimals()) Ownable(msg.sender) {
+        require(_uniswapV2Router != address(0), "TokenBalancer: Invalid Uniswap Router address");
+        require(_vault != address(0), "TokenBalancer: Invalid Vault address");
+        require(_usdc != address(0), "TokenBalancer: Invalid USDC token address");
+
+        uniswapV2Router = IUniswapV2Router02(_uniswapV2Router);
+        underlying = IERC4626(_vault).asset();
+        vault = IERC4626(_vault);
+        usdc = _usdc;
+
+        path = new address[](2);
+        (path[0], path[1]) = (usdc, underlying);
+
+        // Token associations
+        SafeHTS.safeAssociateToken(usdc, address(this));
+        SafeHTS.safeAssociateToken(underlying, address(this));
+        SafeHTS.safeAssociateToken(vault.share(), address(this));
+
+        _createTokenWithContractAsOwner(_name, _symbol, ERC20(underlying));
+    }
+
+    function _createTokenWithContractAsOwner(string memory _name, string memory _symbol, ERC20 _underlying) internal {
+        uint256 supplyKeyType;
+        uint256 adminKeyType;
+
+        IHederaTokenService.KeyValue memory supplyKeyValue;
+        supplyKeyType = supplyKeyType.setBit(4);
+        supplyKeyValue.delegatableContractId = address(this);
+
+        IHederaTokenService.KeyValue memory adminKeyValue;
+        adminKeyType = adminKeyType.setBit(0);
+        adminKeyValue.delegatableContractId = address(this);
+
+        IHederaTokenService.TokenKey[] memory keys = new IHederaTokenService.TokenKey[](2);
+
+        keys[0] = IHederaTokenService.TokenKey(supplyKeyType, supplyKeyValue);
+        keys[1] = IHederaTokenService.TokenKey(adminKeyType, adminKeyValue);
+
+        IHederaTokenService.Expiry memory expiry;
+        expiry.autoRenewAccount = address(this);
+        expiry.autoRenewPeriod = 8000000;
+
+        IHederaTokenService.HederaToken memory newToken;
+        newToken.name = _name;
+        newToken.symbol = _symbol;
+        newToken.treasury = address(this);
+        newToken.expiry = expiry;
+        newToken.tokenKeys = keys;
+        aToken = SafeHTS.safeCreateFungibleToken(newToken, 0, _underlying.decimals());
+        emit CreatedToken(aToken);
+    }
+
+    /*///////////////////////////////////////////////////////////////
+                        DEPOSIT/WITHDRAWAL LOGIC
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @dev Deposits staking token to the Vault and returns shares.
+     * @inheritdoc IAutoCompounder
+     */
+    function deposit(uint256 assets) public override returns (uint256 amountToMint) {
+        require(assets != 0, "AutoCompounder: Invalid assets amount");
+
+        // Calculate aToken amount to mint using exchange rate
+        amountToMint = (assets * 1e18) / exchangeRate();
```

Comment:
why ?
shouldn't the aAsset have the same number of decimal points as the original ?